sdks/wasm/bindings.cs

```diff
@@ -79,69 +104,68 @@ static int UnBindJSObject(int js_id)
 
         }
 
-        static int BindJSObject(JSObject obj)
-        {
-
-            int js_id = obj.JSHandle;
-            if (js_id <= 0)
-                throw new JSException($"Invalid JS Object Handle {js_id}");
-
-            if (bound_objects.ContainsKey(js_id))
-                obj = bound_objects[js_id];
-            else
-                bound_objects[js_id] = obj;
-
-            return (int)(IntPtr)obj.Handle;
-        }
-
-        static int UnBindJSObjectAndFree(int js_id)
+        static void UnBindJSObjectAndFree(int js_id)
         {
             if (bound_objects.ContainsKey(js_id))
             {
-                var obj = bound_objects[js_id];
+                bound_objects[js_id].RawObject = null;
+                JSObject obj = bound_objects[js_id];
                 bound_objects.Remove(js_id);
-                var gCHandle = obj.Handle;
-                obj.Handle.Free();
                 obj.JSHandle = -1;
                 obj.RawObject = null;
-                return (int)(IntPtr)gCHandle;
+                obj.Handle.Free();
+
             }
-            return 0;
 
         }
 
 
-        static int UnBindRawJSObjectAndFree(int gcHandle)
+        static void UnBindRawJSObjectAndFree(int gcHandle)
         {
 
             GCHandle h = (GCHandle)(IntPtr)gcHandle;
             JSObject obj = (JSObject)h.Target;
             if (obj != null && obj.RawObject != null)
             {
-                var raw_obj = obj.RawObject;
-                if (raw_to_js.ContainsKey(raw_obj))
-                {
-                    raw_to_js.Remove(raw_obj);
-                }
+                raw_to_js.Remove(obj.RawObject);
 
-                obj.Dispose();
+                int exception;
+                ReleaseHandle(obj.JSHandle, out exception);
+                if (exception != 0)
+                    throw new JSException($"Error releasing handle on (js-obj js '{obj.JSHandle}' mono '{(IntPtr)obj.Handle} raw '{obj.RawObject != null})");
+                
+                // Calling Release Handle above only removes the reference from the JavaScript side but does not 
+                // release the bridged JSObject associated with the raw object so we have to do that ourselves.
+                obj.JSHandle = -1;
+                obj.RawObject = null;
 
-                var gCHandle = obj.Handle;
                 obj.Handle.Free();
-                obj.JSHandle = -1;
-                return (int)(IntPtr)gCHandle;
             }
-            return 0;
 
         }
 
         public static void FreeObject (object obj)
         {
-            int exception;
-            Runtime.ReleaseObject(obj, out exception);
-            if (exception != 0)
-                throw new JSException($"Error releasing object on (raw-obj)");
-   
+            if (raw_to_js.ContainsKey(obj))
```

Comment:
```suggestion
            if (raw_to_js.TryGetValue(obj, out JSObject jsobj))
```

Reply:
Here if I do the `TryGetValue` then I have to reference it again later with `raw_to_js[obj].RawObject = null;` anyway to circumvent an intermittent `Uncaught RuntimeError: memory access out of bounds` when GC runs.